engine/collection/compliance/core.go

```diff
@@ -267,69 +312,111 @@ func (c *Core) processBlockProposal(proposal *messages.ClusterBlockProposal) err
 		Logger()
 	log.Info().Msg("processing block proposal")
 
+	hotstuffProposal := model.ProposalFromFlow(header)
+	err := c.validator.ValidateProposal(hotstuffProposal)
+	if err != nil {
+		if model.IsInvalidBlockError(err) {
+			return engine.NewInvalidInputErrorf("invalid block proposal: %w", err)
+		}
+		if errors.Is(err, model.ErrViewForUnknownEpoch) {
+			// We have received a proposal, but we don't know the epoch its view is within.
+			// We know:
+			//  - the parent of this block is valid and was appended to the state (ie. we knew the epoch for it)
+			//  - if we then see this for the child, one of two things must have happened:
+			//    1. the proposer malicious created the block for a view very far in the future (it's invalid)
+			//      -> in this case we can disregard the block
+			//    2. no blocks have been finalized within the epoch commitment deadline, and the epoch ended
+			//       (breaking a critical assumption - see EpochCommitSafetyThreshold in protocol.Params for details)
+			//      -> in this case, the network has encountered a critical failure
+			//  - we assume in general that Case 2 will not happen, however we cannot prove Case 1, therefore we can discard this proposal
+			return engine.NewUnverifiableInputError("cannot validate proposal with view from unknown epoch: %w", err)
```

Comment:
The cluster committee never actually returns `ErrViewForUnknownEpoch`. So I think we should treat this as an unexpected error here. 

We can remove the comment as well, since that is only applicable to consensus nodes. 
```suggestion
			// The cluster committee never returns ErrViewForUnknownEpoch, therefore this case
			// is an unexpected error in cluster consensus.
			return fmt.Errorf("unexpected error: cluster committee reported unknown epoch : %w", err)
```